### PR TITLE
added portfolio / cash management objects

### PR DESCRIPTION
**Pull request by juggler99**
Still needs some work but it is a start

### CODE REVIEW

File: epython/portfolio_management/cash/interest_rate.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    def sql(self, dt, ccy):
+        dt = self._ensure_datetime(dt)
+
+        kwargs = {'dt': dt, 'ccy': ccy}
+
+        for k, v in self.sql_kwargs().items():
+            kwargs[k] = v
+
+        return self.sql_template().format(**kwargs)
```
Comment (tayglobal) [inline epython/portfolio_management/cash/interest_rate.py]: Shouldn't expose SQL logic in the interest rate class

File: epython/portfolio_management/cash/interest_rate.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        return {
+            'QueryString': self.sql(dt, ccy),
+            'QueryExecutionContext': {
+                'Database': self.database(),
+            },
+            'ResultConfiguration': {
+                'OutputLocation': self.res_s3_path(),
+            },
+            'WorkGroup': 'primary'
+        }
```
Comment (tayglobal) [inline epython/portfolio_management/cash/interest_rate.py:72]: Shoudn't expose SQL logic in Interest rate class

File: epython/portfolio_management/cash/interest_rate.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    FROM interest_rate
+    WHERE pair = '{pair}'
+            AND year = '{dt:%Y}'
+SELECT *
+FROM t
+WHERE dt = date = '{dt:%Y-%m-%d}'        
+'''
+
+
+class InterestRate(SQLMixin, kydb.DbObj):
```
Comment (tayglobal) [inline epython/portfolio_management/cash/interest_rate.py:16]: Need only be a subclass of kydb.DbObj

File: epython/portfolio_management/utils/sql.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    def sql_kwargs(self) -> dict:
+        return {}
+
+    @staticmethod
+    def _ensure_datetime(dt: Union[date, datetime]) -> datetime:
+        if isinstance(dt, date):
+            return datetime.combine(dt, datetime.min.time())
+
+        return dt
+
```
Comment (tayglobal) [inline epython/portfolio_management/utils/sql.py:24]: We aleady have a SQL Mixin. Seemed liek duplicated logic

**Review by tayglobal — CHANGES_REQUESTED** [review]
Seemed the TSDB logic got exposed into the InterestRate class and there is a SQLMixin class that already exists.

The time series data access is already abstracted away into TSDB. Let's not couple those implementation into business logic. Interest rate class shoudln't care if it's taking data from Athena, dynamo, KDB+, s3, etc...

File: epython/portfolio_management/cash/interest_rate.py (hunk 1 above)
Comment (juggler99) [reply epython/portfolio_management/cash/interest_rate.py]: Agree, it is a left over form when I copied the initial class, I will get rid of it. As mentioned in slack, this is the class I spent less time on, I was trying to get everything else to a semi working stage. Will have a look.

Comment (tayglobal) [reply epython/portfolio_management/cash/interest_rate.py]: Cool. I'll spend more time tomorrow going through the PR.

Comment (juggler99) [reply epython/portfolio_management/cash/interest_rate.py]: AT the moment interest rate is hardcoded in account eod process (interest_rate)

Comment (juggler99) [reply epython/portfolio_management/cash/interest_rate.py]: Forgot to add, if you look at test_portfolio, that loads all object types to a portfolio

**Pull request closed without merging**